Only copy Model sandbox to install path on visionOS

pvollan opened the pull request:
#### 240c462c7d4fe10b6b68f3f20f8a9e7a25e2f4c8
<pre>
Only copy Model sandbox to install path on visionOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=297856">https://bugs.webkit.org/show_bug.cgi?id=297856</a>
<a href="https://rdar.apple.com/problem/159100251">rdar://problem/159100251</a>

Reviewed by Elliott Williams.

This sandbox is only used on visionOS. Also, only copy the developments sandboxes to the install path on iOS,
since they are only intended to be used there at the moment.

* Source/WebKit/DerivedSources.make:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/299283@main">https://commits.webkit.org/299283@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4da06791f9f6d1c5e8bddb5be6d7d073be9e9f3a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118468 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/38149 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28800 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124639 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70527 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/947c8642-3201-41c1-8635-7e3de9804a37) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38845 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46731 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89911 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/59493 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/c65b9fad-cd50-4bcd-b6ee-c478e1d30e58) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121421 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30957 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/106226 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70396 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/05b66183-4e13-4bb9-bd21-3aff25036052) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/30015 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/68296 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100394 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24526 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127706 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45375 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34233 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/98577 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45739 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102445 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98362 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43780 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21774 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/41873 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18877 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/45245 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50923 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44708 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/48055 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46395 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/WebKit/DerivedSources.make:
```diff
@@ -452,20 +452,35 @@ WEBPUSHD_SANDBOX_PROFILE = \
+	com.apple.WebKit.Networking.Development.sb \
+	com.apple.WebKit.WebContent.Development.sb
+else
+SANDBOX_PROFILES_EMBEDDED_DEVELOPMENT =
```
nit: no need to define an empty variable
```suggestion
```

emw-apple commented on Source/WebKit/DerivedSources.make:
```diff
@@ -452,20 +452,35 @@ WEBPUSHD_SANDBOX_PROFILE = \
+SANDBOX_PROFILES_EMBEDDED_DEVELOPMENT =
+endif
+
+SANDBOX_PROFILES_EMBEDDED_MODEL =
```
nit: no need to define an empty variable
```suggestion
```

emw-apple commented on Source/WebKit/DerivedSources.make:
```diff
@@ -452,20 +452,35 @@ WEBPUSHD_SANDBOX_PROFILE = \
 	com.apple.WebKit.WebContent.sb \
-	com.apple.WebKit.WebContent.Development.sb
+	$(SANDBOX_PROFILES_EMBEDDED_MODEL) \
+	$(SANDBOX_PROFILES_EMBEDDED_DEVELOPMENT)
```
Instead of splitting out a separate variables above, I think it would be more natural to use `+=` like this:

```make
SANDBOX_PROFILES_EMBEDDED = \
    com.apple.WebKit.adattributiond.sb \
    ...

ifneq ($(filter xr%,$(WK_PLATFORM_NAME)),)
SANDBOX_PROFILES_EMBEDDED += com.apple.WebKit.Model.sb
endif

ifeq ($(WK_PLATFORM_NAME), iphoneos)
SANDBOX_PROFILES_EMBEDDED += com.apple.WebKit.GPU.Development.sb \
	com.apple.WebKit.Networking.Development.sb \
	com.apple.WebKit.WebContent.Development.sb
endif
```

This also uses [`$(filter ...)`](https://www.gnu.org/software/make/manual/html_node/Text-Functions.html#index-filtering-words) to check whether WK_PLATFORM_NAME starts with `xr`, so you can combine the visionOS device/simulator checks into one if statement.

emw-apple commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -20325,7 +20325,7 @@
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "echo \"Preprocessing sandbox\"\nScripts/generate-derived-sources.sh sandbox-profiles-ios\nmkdir -p ${DSTROOT}/${INSTALL_PATH}\ncp ${BUILT_PRODUCTS_DIR}/DerivedSources/WebKit/com.apple.WebKit.adattributiond.sb ${DSTROOT}/${INSTALL_PATH}\ncp ${BUILT_PRODUCTS_DIR}/DerivedSources/WebKit/com.apple.WebKit.webpushd.sb ${DSTROOT}/${INSTALL_PATH}\ncp ${BUILT_PRODUCTS_DIR}/DerivedSources/ ...
+			shellScript = "echo \"Preprocessing sandbox\"\n\nScripts/generate-derived-sources.sh sandbox-profiles-embedded\nmkdir -p ${DSTROOT}/${INSTALL_PATH}\n\ncp ${BUILT_PRODUCTS_DIR}/DerivedSources/WebKit/com.apple.WebKit.adattributiond.sb ${DSTROOT}/${INSTALL_PATH}\ncp ${BUILT_PRODUCTS_DIR}/DerivedSources/WebKit/com.apple.WebKit.webpushd.sb ${DSTROOT}/${INSTALL_PATH}\ncp ${BUILT_PRODUCTS_DIR}/Derive ...
```
Here's the diff:
```diff
< Scripts/generate-derived-sources.sh sandbox-profiles-ios
---
> 
> Scripts/generate-derived-sources.sh sandbox-profiles-embedded
3a5
> 
7d8
< cp ${BUILT_PRODUCTS_DIR}/DerivedSources/WebKit/com.apple.WebKit.GPU.Development.sb ${DSTROOT}/${INSTALL_PATH}
9d9
< cp ${BUILT_PRODUCTS_DIR}/DerivedSources/WebKit/com.apple.WebKit.Networking.Development.sb ${DSTROOT}/${INSTALL_PATH}
12d11
< cp ${BUILT_PRODUCTS_DIR}/DerivedSources/WebKit/com.apple.WebKit.WebContent.Development.sb ${DSTROOT}/${INSTALL_PATH}
17a17,22
> if [[ -e ${BUILT_PRODUCTS_DIR}/DerivedSources/WebKit/com.apple.WebKit.GPU.Development.sb ]]; then
>     cp ${BUILT_PRODUCTS_DIR}/DerivedSources/WebKit/com.apple.WebKit.GPU.Development.sb ${DSTROOT}/${INSTALL_PATH}
>     cp ${BUILT_PRODUCTS_DIR}/DerivedSources/WebKit/com.apple.WebKit.Networking.Development.sb ${DSTROOT}/${INSTALL_PATH}
>     cp ${BUILT_PRODUCTS_DIR}/DerivedSources/WebKit/com.apple.WebKit.WebContent.Development.sb ${DSTROOT}/${INSTALL_PATH}
> fi
> 
```

Could you add a `set -e` to the top so that a failure to run `make` fails the build? Seems like we should have always done that. Otherwise LGTM.

pvollan replied:
Ah, good catches, will fix!

Thanks for reviewing!

emw-apple reviewed (approved):
I have a couple requests, but the logic makes sense, so feel free to land once you've addressed them.

pvollan committed:
Only copy Model sandbox to install path on visionOS
https://bugs.webkit.org/show_bug.cgi?id=297856
rdar://problem/159100251

Reviewed by Elliott Williams.

This sandbox is only used on visionOS. Also, only copy the developments sandboxes to the install path on iOS,
since they are only intended to be used there at the moment.

* Source/WebKit/DerivedSources.make:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/299283@main
